### PR TITLE
[Sprint 50]XD-3151 Updated Modules.asciidoc

### PR DESCRIPTION
**Pull request by dturanski**

### CODE REVIEW

File: src/docs/asciidoc/Modules.asciidoc [hunk 1]
```diff
@@ -105,40 +121,44 @@ buildscript {
...
+==== Module dependency management
+
+Runtime issues may arise if module dependencies conflict with those on the Spring XD classpath. This happens when a module declares a dependency using a different version of a jar provided by the Spring XD runtime. If the versions are the same, the build tools will ensure that these dependencies will not be exported to the module jar. However if the versions declared in your build are different, the dependencies will be exported and loaded by the module class loader. The tools are designed to permit you to override the version, but generally we don't recommend this unless you have a specific requirement to do so. If there are conflicting dependencies, your module may or may not work. Symptoms of version conflicts include `ClassDefNotFoundException` or `NoSuchMethodError` when you deploy the module.  If you encounter such errors, you should manually check the contents of `$XD_INSTALL_DIR/xd/lib` against the contents of the module jar, and make the necessary changes to your build script. Compile time dependencies must be declared in any case. If they are also on the Spring XD classpath they should be declared as `provided`.
```
Comment (ericbottard) [inline src/docs/asciidoc/Modules.asciidoc]: > if the versions are different ...

I don't think this is true, is it really?

Comment (dturanski) [inline src/docs/asciidoc/Modules.asciidoc]: I should clarify this. It's true for Gradle, not Maven which is explained further down.

**Comment by dturanski** [comment]
Another improvement IMO is to organize this into separate sections, Building with Maven and Building with Gradle. I think most users would find this more useful.

**Comment by ericbottard** [comment]
Current state of this PR merged as a06033e53fcf8413789351299202715b055d9bf1 sorry. Feel free to either
* close this and issue another PR
* append more commits, will rebase and cherry-pick as needed

**Comment by dturanski** [comment]
closed; Issued a new PR https://github.com/spring-projects/spring-xd/pull/1697

**Pull request closed without merging**